### PR TITLE
Update react dependencies

### PR DESCRIPTION
**Pull request by iwysiu**
I needed to update the react packages to do local development with athena, since the packages there are updated. I assume this would be true for the other datasources as well.

### CODE REVIEW

File: src/sql/QueryEditor/QueryEditorHeader.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ export interface Props<
   extraHeaderElementLeft?: JSX.Element;
   extraHeaderElementRight?: JSX.Element;
   enableRunButton: boolean;
-  cancel?: (target: TQuery) => void;
```
Comment (sarahzinger) [inline src/sql/QueryEditor/QueryEditorHeader.tsx:17]: just double checking this change is intentional? (assuming it is? just seems like a breaking change i think so wanted to double check)

Comment (iwysiu) [reply src/sql/QueryEditor/QueryEditorHeader.tsx:17]: we need it to satisfy the api that we're using

Comment (iwysiu) [reply src/sql/QueryEditor/QueryEditorHeader.tsx]: retried it and i ended up not need it!

**Review by kevinwcyu — APPROVED** [review]

File: src/sql/QueryEditor/QueryEditorHeader.tsx (hunk 1 above)
Comment (iwysiu) [reply src/sql/QueryEditor/QueryEditorHeader.tsx]: i take that back, i did need it to be defined but i made it not a breaking change.

**Pull request merged**